x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,157 @@
+
+	// Check that the validity of the misbehaviour
+	if err := k.clientKeeper.CheckMisbehaviourAndUpdateState(ctx, misbehaviour); err != nil {
+		logger.Info("Misbehaviour rejected", err.Error())
+
+		return err
+	}
+
+	// Assign the Tendermint client misbehaviour concrete type
+	tmMisbehaviour := misbehaviour.(*ibctmtypes.Misbehaviour)
+
+	// Since the misbehaviour packet was received within the trusting period
+	// w.r.t to the last trusted consensus it entails that the infraction age
+	// isn't too old. see ibc-go/modules/light-clients/07-tendermint/types/misbehaviour_handle.go
+
+	// construct a ligth client attack evidence
+	evidence, err := k.ConstructLightClientEvidence(ctx, *tmMisbehaviour)
+	if err != nil {
+		return err
+	}
+
+	logger.Info("ConstructLightClientEvidence", fmt.Sprintf("%#+v\n", evidence))
+
+	// jail and tombstone the byzantine validators
+	for _, v := range evidence.ByzantineValidators {
+		// convert consumer consensus address
+		consuAddr := sdk.ConsAddress(v.Address.Bytes())
+		provAddr := k.GetProviderAddrFromConsumerAddr(ctx, tmMisbehaviour.Header1.Header.ChainID, types.NewConsumerConsAddress(consuAddr))
+		k.stakingKeeper.ValidatorByConsAddr(ctx, consuAddr)
+		val, ok := k.stakingKeeper.GetValidatorByConsAddr(ctx, provAddr.Address)
+
+		if !ok || val.IsUnbonded() {
+			logger.Error("validator not found or is unbonded", provAddr.String())
+			continue
+		}
+
+		// jail validator if not already
+		if !val.IsJailed() {
+			k.stakingKeeper.Jail(ctx, provAddr.Address)
+		}
+
+		// tombstone validator if not already
+		if !k.slashingKeeper.IsTombstoned(ctx, provAddr.Address) {
+			k.slashingKeeper.Tombstone(ctx, provAddr.Address)
+		}
+
+		// update jail time to end after double sign jail duration
+		k.slashingKeeper.JailUntil(ctx, provAddr.Address, evidencetypes.DoubleSignJailEndTime)
+	}
+
+	logger.Info(
+		"confirmed equivocation",
+		"byzantine validators", evidence.ByzantineValidators,
+	)
+
+	return nil
+}
+
+// ConstructLightClientEvidence constructs and returns a CometBFT Ligth Client Attack(LCA) evidence struct
+// from the given misbehaviour
+func (k Keeper) ConstructLightClientEvidence(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) (*tmtypes.LightClientAttackEvidence, error) {
+	// construct the trusted and conflicetd ligth blocks
+	trusted, err := headerToLightBlock(*misbehaviour.Header1)
+	if err != nil {
+		return nil, err
+	}
+	conflicted, err := headerToLightBlock(*misbehaviour.Header2)
+	if err != nil {
+		return nil, err
+	}
+
+	// get common header using the IBC misbehaviour
+	commonHeight, commonTs, commonValset, err := k.GetTrustedInfoFromMisbehaviour(ctx, misbehaviour)
+	if err != nil {
+		return nil, err
+	}
+
+	// construct the LCA evidence by copying the CometBFT constructor
+	// see newLightClientAttackEvidence() in tendermint/light/detector.go
+	ev := tmtypes.LightClientAttackEvidence{
+		ConflictingBlock: conflicted,
+	}
+
+	if ev.ConflictingHeaderIsInvalid(trusted.Header) {
```

Comment:
my understanding is that regardless of the outcome of this function,
the height, timestamp and totalvotingpower come from `trusted` -
either they are taken directly from the trusted block, or they are the same in both anyways.

Is this if/else even needed? If yes, could you clarify in a comment?